android/src/main/java/com/swmansion/rnscreens/ScreenFragment.java

```diff
@@ -65,45 +66,59 @@ public Screen getScreen() {
   }
 
   public void onContainerUpdate() {
-    if (!hasChildScreenWithConfig(getScreen())) {
-      // if there is no child with config, we look for a parent with config to set the orientation
-      ScreenStackHeaderConfig config = findHeaderConfig();
-      if (config != null && config.getScreenFragment().getActivity() != null) {
-        config.getScreenFragment().getActivity().setRequestedOrientation(config.getScreenOrientation());
-      }
-    }
+   updateWindowTraits();
   }
 
-  private @Nullable ScreenStackHeaderConfig findHeaderConfig() {
+  private void updateWindowTraits() {
+    Screen screen = getScreen();
+    Activity activity = tryGetActivity();
+    ReactContext context = tryGetContext();
+    ScreenWindowTraits.setOrientation(ScreenWindowTraits.findScreenForTrait(screen, "orientation"), activity);
```

Comment:
I know why you wrote these lines this way but it would be nice to hide the implementation details in the `ScreenWindowTraits` class i.e. calling `findScreenForTrait` inside these methods if it's possible of course.

Reply:
It is not possible because `ScreenStackHeaderConfig` also uses the `ScreenWindowTraits.set...` methods and we don't want to apply the `ScreenWindowTraits.findScreenForTrait` method logic there.